server/log_rpc_server_test.go

```diff
@@ -350,95 +348,90 @@ func TestQueueLeavesBeginFailsCausesError(t *testing.T) {
+			commitErr: errors.New("Commit() error"),
+		},
+		{
+			// Test normal case where a root is returned correctly.
+			req:         getLogRootRequest1,
+			wantRoot:    trillian.GetLatestSignedLogRootResponse{SignedLogRoot: &signedRoot1},
+			storageRoot: signedRoot1,
 		},
-		func(s *TrillianLogRPCServer) error {
-			_, err := s.GetLatestSignedLogRoot(context.Background(), &getLogRootRequest1)
-			return err
-		})
-
-	test.executeCommitFailsTest(t, getLogRootRequest1.LogId)
-}
-
-func TestGetLatestSignedLogRootInvalidLogId(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	test := newParameterizedTest(ctrl, "LatestSignedLogRoot", readOnly,
-		func(t *storage.MockLogTreeTX) {},
-		func(s *TrillianLogRPCServer) error {
-			_, err := s.GetLatestSignedLogRoot(context.Background(), &getLogRootRequest2)
-			return err
-		})
-
-	test.executeInvalidLogIDTest(t, true /* snapshot */)
-}
-
-func TestGetLatestSignedLogRoot(t *testing.T) {
-	ctrl := gomock.NewController(t)
-	defer ctrl.Finish()
-
-	mockStorage := storage.NewMockLogStorage(ctrl)
-	mockTx := storage.NewMockLogTreeTX(ctrl)
-	mockStorage.EXPECT().SnapshotForTree(gomock.Any(), getLogRootRequest1.LogId).Return(mockTx, nil)
-	mockTx.EXPECT().LatestSignedLogRoot(gomock.Any()).Return(signedRoot1, nil)
-	mockTx.EXPECT().Commit().Return(nil)
-	mockTx.EXPECT().Close().Return(nil)
-
-	registry := extension.Registry{
-		AdminStorage: mockAdminStorage(ctrl, getLogRootRequest1.LogId),
-		LogStorage:   mockStorage,
 	}
-	server := NewTrillianLogRPCServer(registry, fakeTimeSource)
 
-	resp, err := server.GetLatestSignedLogRoot(context.Background(), &getLogRootRequest1)
-	if err != nil {
-		t.Fatalf("Failed to get log root: %v", err)
-	}
+	for _, test := range tests {
+		mockStorage := storage.NewMockLogStorage(ctrl)
+		mockTx := storage.NewMockLogTreeTX(ctrl)
+		if !test.noSnap {
+			mockStorage.EXPECT().SnapshotForTree(gomock.Any(), test.req.LogId).Return(mockTx, test.snapErr)
+		}
+		if !test.noRoot {
+			mockTx.EXPECT().LatestSignedLogRoot(gomock.Any()).Return(test.storageRoot, test.rootErr)
+		}
+		if !test.noCommit {
+			mockTx.EXPECT().Commit().Return(test.commitErr)
+		}
+		if !test.noClose {
+			mockTx.EXPECT().Close().Return(nil)
+		}
 
-	if !proto.Equal(&signedRoot1, resp.SignedLogRoot) {
-		t.Fatalf("Log root proto mismatch:\n%v\n%v", signedRoot1, resp.SignedLogRoot)
+		registry := extension.Registry{
+			AdminStorage: mockAdminStorage(ctrl, test.req.LogId),
+			LogStorage:   mockStorage,
+		}
+		s := NewTrillianLogRPCServer(registry, fakeTimeSource)
+		resp, err := s.GetLatestSignedLogRoot(context.Background(), &test.req)
+		if test.wantErr {
+			if err == nil {
+				t.Errorf("GetLatestSignedLogRoot(%+v)=nil, want: err", test.req)
+			}
+		} else {
+			if err != nil {
+				t.Errorf("GetLatestSignedLogRoot(%+v)=%v, want: nil", test.req, err)
+			}
+			// Ensure we got the expected root back.
+			if got, want := resp, &test.wantRoot; !proto.Equal(got, want) {
```

Comment:
rename `resp` to `got` and you could drop one of the scope-locals here; if it were me, I'd drop the other one too as `test.wantRoot` already has the magic "want" substring

Reply:
Done.